Updated dependencies and fixed imports

damc-dev opened the pull request:
Hey Stephanie, I made the following changes and was able to get it running on my laptop with Nixpacks, give it a try!

If these changes don't work in railway you probably need to contact them to see if they can help.

stephanievelez commented:
Good morning David,

So how were you able to know which dependencies to remove as well as
knowing to edit the fastbook import?

On Tue, Jan 23, 2024 at 10:10 PM David McElligott ***@***.***>
wrote:

> Hey Stephanie, I made the following changes and was able to get it running
> on my laptop with Nixpacks, give it a try!
>
> If these changes don't work in railway you probably need to contact them
> to see if they can help.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/stephanievelez/porfolio-railway/pull/1
> Commit Summary
>
>    - 740b732
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/commits/740b732f71492a21034bc9725b0a22b778dc1d6f>
>    build: commented out unused dependencies
>    - 292e97d
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/commits/292e97dac1122c2bb6a98a385ce1bc644ba1d5b5>
>    fix: removed fastbook imports
>    - a152ed5
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/commits/a152ed58ee5ac69c9569157d2032b3f280b10689>
>    fix: added missing import and removed unnecessary heroku files"
>    - 2bdca9c
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/commits/2bdca9c46a4764c2f8a42be40db888a73aadc3aa>
>    fix: corrected import
>    - f6874a4
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/commits/f6874a402a5da8866fbd6e698e2892054dffb484>
>    build: removed more dependencies
>    - bc0c2f7
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/commits/bc0c2f7f97fdd106dfd50e884cdbbd06c91fdbae>
>    docs: Added commands to run with Nixpack locally
>
> File Changes
>
> (6 files <https://github.com/stephanievelez/porfolio-railway/pull/1/files>
> )
>
>    - *M* README.md
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>    (17)
>    - *M* alzheimerModel/model/model.py
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-7e7de4cb6afbf20e29532b7a6767f023ae8de0ab13e3027f8a49ac915e0c5ace>
>    (6)
>    - *M* alzheimerModel/views.py
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-57154cab622d84bc4c77488282347e76cb27e1418b0d2d48698e4b750b59fbf0>
>    (5)
>    - *D* learning_log/Procfile
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-4eec9ed100f009d1eaf848a68a4e0248d7734226f291e334c79f134343eeee92>
>    (5)
>    - *D* learning_log/runtime.txt
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-2c035a803d30175470a0b5f5dfdf45ba8bb43c37925a321111d324e8d5142054>
>    (5)
>    - *M* requirements.txt
>    <https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552>
>    (328)
>
> Patch Links:
>
>    - https://github.com/stephanievelez/porfolio-railway/pull/1.patch
>    - https://github.com/stephanievelez/porfolio-railway/pull/1.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/stephanievelez/porfolio-railway/pull/1>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABLBT6PJNJXCKVEF4KIHXTLYQCCSZAVCNFSM6AAAAABCICSCQ2VHI2DSMVQWIX3LMV43ASLTON2WKOZSGA4TOMZZGIYDQNQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

damc-dev commented:
Hey Stephanie,

Sorry for the delay in getting back to you!  I went through your code to
see what dependencies your Python files are importing and commented out all
the others.

I did have to try and build and run the application (with Nixpacks, so
dependencies weren't cached) a couple of times, and it would fail because I
had deleted a dependency that was required, so I would re-add the missing
one to the requirements.txt file and then try again.

As for the fastbook code, I googled it to see what it does and found it was
used for initializing something in Jupyter Notebooks; since the application
isn't dependent on Jupyter Notebooks at all, I knew I could remove it.

Let me know if you have any other questions about the dependencies!


Thanks,
David

On Wed, Jan 24, 2024 at 9:07 AM Dr. Stephanie Velez <
***@***.***> wrote:

> Good morning David,
>
> So how were you able to know which dependencies to remove as well as
> knowing to edit the fastbook import?
>
> On Tue, Jan 23, 2024 at 10:10 PM David McElligott ***@***.***>
> wrote:
>
> > Hey Stephanie, I made the following changes and was able to get it
> running
> > on my laptop with Nixpacks, give it a try!
> >
> > If these changes don't work in railway you probably need to contact them
> > to see if they can help.
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/stephanievelez/porfolio-railway/pull/1
> > Commit Summary
> >
> > - 740b732
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/commits/740b732f71492a21034bc9725b0a22b778dc1d6f>
>
> > build: commented out unused dependencies
> > - 292e97d
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/commits/292e97dac1122c2bb6a98a385ce1bc644ba1d5b5>
>
> > fix: removed fastbook imports
> > - a152ed5
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/commits/a152ed58ee5ac69c9569157d2032b3f280b10689>
>
> > fix: added missing import and removed unnecessary heroku files"
> > - 2bdca9c
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/commits/2bdca9c46a4764c2f8a42be40db888a73aadc3aa>
>
> > fix: corrected import
> > - f6874a4
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/commits/f6874a402a5da8866fbd6e698e2892054dffb484>
>
> > build: removed more dependencies
> > - bc0c2f7
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/commits/bc0c2f7f97fdd106dfd50e884cdbbd06c91fdbae>
>
> > docs: Added commands to run with Nixpack locally
> >
> > File Changes
> >
> > (6 files <
> https://github.com/stephanievelez/porfolio-railway/pull/1/files>
> > )
> >
> > - *M* README.md
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>
> > (17)
> > - *M* alzheimerModel/model/model.py
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-7e7de4cb6afbf20e29532b7a6767f023ae8de0ab13e3027f8a49ac915e0c5ace>
>
> > (6)
> > - *M* alzheimerModel/views.py
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-57154cab622d84bc4c77488282347e76cb27e1418b0d2d48698e4b750b59fbf0>
>
> > (5)
> > - *D* learning_log/Procfile
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-4eec9ed100f009d1eaf848a68a4e0248d7734226f291e334c79f134343eeee92>
>
> > (5)
> > - *D* learning_log/runtime.txt
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-2c035a803d30175470a0b5f5dfdf45ba8bb43c37925a321111d324e8d5142054>
>
> > (5)
> > - *M* requirements.txt
> > <
> https://github.com/stephanievelez/porfolio-railway/pull/1/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552>
>
> > (328)
> >
> > Patch Links:
> >
> > - https://github.com/stephanievelez/porfolio-railway/pull/1.patch
> > - https://github.com/stephanievelez/porfolio-railway/pull/1.diff
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/stephanievelez/porfolio-railway/pull/1>, or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ABLBT6PJNJXCKVEF4KIHXTLYQCCSZAVCNFSM6AAAAABCICSCQ2VHI2DSMVQWIX3LMV43ASLTON2WKOZSGA4TOMZZGIYDQNQ>
>
> > .
> > You are receiving this because you are subscribed to this thread.Message
> > ID: ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/stephanievelez/porfolio-railway/pull/1#issuecomment-1908320828>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5OKTKOSVWUQWDSQB53KTLYQEPSPAVCNFSM6AAAAABCICSCQ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBYGMZDAOBSHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
